README.md
<p align="center">
   <a href="https://themeselection.com/item/sneat-mui-react-nextjs-admin-template" target="_blank">
      <img src="https://cdn.jsdelivr.net/gh/themeselection/ts-assets/sneat/sneat-mui-nextjs-admin-template/logo/logo.png" alt="sneat-logo" width="40" height="auto">
   </a>
</p>

<h1 align="center">
   <a href="https://themeselection.com/item/sneat-mui-react-nextjs-admin-template" target="_blank" align="center">
      Sneat - MUI React NextJS Admin Template
   </a>
</h1>

<p align="center">Most Powerful & Comprehensive MUI React NextJS Admin Dashboard Template built for developers!</p>

<p align="center">
  <a href="https://twitter.com/Theme_Selection" target="_blank">
     <img alt="Twitter Follow" src="https://img.shields.io/twitter/follow/Theme_Selection">
  </a>
</p>

<p align="center">
   <a href="https://themeselection.com/item/sneat-mui-react-nextjs-admin-template" target="_blank">
      <img src="https://cdn.jsdelivr.net/gh/themeselection/ts-assets/sneat/sneat-mui-nextjs-admin-template/banner/banner.png" alt="Sneat - MUI React NextJS Admin Template">
   </a>
</p>

## Introduction

If you’re a developer looking for most Powerful & comprehensive **MUI React NextJS Admin Dashboard Template** built for developers, rich with features, and highly customizable, look no further than sneat. We’ve followed the highest industry standards to bring you the very best admin template that is not only easy to use but highly scalable. Offering ultimate convenience and flexibility, you’ll be able to build whatever application you want with very little hassle.

Build premium quality applications with ease. Use our innovative admin template to create eye-catching, high-quality WebApps. Your apps will be completely responsive, ensuring they’ll look stunning and function flawlessly on desktops, tablets, and mobile devices.

[View Demo](https://demos.themeselection.com/sneat-mui-react-nextjs-admin-template/demo-1/)

## Installation ⚒️

Please [visit](https://demos.themeselection.com/sneat-mui-react-nextjs-admin-template/documentation/guide/overview/installation.html) our docs for installation guide.

## Documentation 📜

Check out our live [Documentation](https://demos.themeselection.com/sneat-mui-react-nextjs-admin-template/documentation/)

## Learn More

To learn more about Next.js, take a look at the following resources:

- [Next.js Documentation](https://nextjs.org/docs) - learn about Next.js features and API.
- [Learn Next.js](https://nextjs.org/learn) - an interactive Next.js tutorial.

You can check out [the Next.js GitHub repository](https://github.com/vercel/next.js/) - your feedback and contributions are welcome!

## Deploy on Vercel

The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.

Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.

## Support 👨‍💻

We use GitHub issues as support tickets to manage Item support.

1. In order to raise the GitHub issue, you must have our Premium Item repository access. [Get repo access](https://tools.themeselection.com/github/github-access).
2. Make sure you use our [Raise Support](https://tools.themeselection.com/github/raise-support) form to create an issue.

## Why forking is disabled? 🔒

You will lose repository access when your support expires. If you have forked the repo and you lose the access to our repo then your fork also got deleted by GitHub.

Hence, as precautions we disabled forking our repo.

**Solution**

However, you can still keep your repo even after your support expires by setting our repo as upstream:

1. Create a new **private repo** in your GitHub. Make sure it's a private repo. You can't share our code publicly as per license.
2. Clone our repo: `git clone https://github.com/pixinvent/vuexy-nextjs-admin-template`
3. Navigate to cloned directory
4. Remove our repo as origin by running command: `git remote remove origin`
5. Add your newly created repo as origin by running command: `git remote add origin YOUR_NEWLY_CREATE_REPO_GIT_URL`
6. Add our repo as upstream by running command: `git remote add upstream https://github.com/pixinvent/vuexy-nextjs-admin-template`
7. Push the code to GitHub by running command: `git push -u origin main`

Now, whenever you want to pull the latest changes from our repo just pull the changes by running command: `git pull upstream main`

Cheers 🥂

## Contributing 🦸

Contributions are always welcome and recommended! Here is how:

- Clone our repo `git clone https://github.com/pixinvent/vuexy-nextjs-admin-template`
- create a new branch based on branch you want to make Pull request to. For example if you want to make pull request to main branch, create new branch based on main branch.
- Push your changes to new branch
- Create pull request

### Contribution Requirements 🧰

- When you contribute, you agree to give a non-exclusive license to ThemeSelection to use that contribution in any context as we (ThemeSelection) see appropriate.
- If you use content provided by another party, it must be appropriately licensed using an open source license.
- Contributions are only accepted through Github pull requests.
- Finally, contributed code must work in all supported browsers.

## Creators 😇

- [ThemeSelection](https://themeselection.com)

## Changelog 📆

Please refer to the [CHANGELOG](CHANGELOG.md) file. We will add a detailed release notes to each new release.

## Social Media 🌍

- Twitter : [https://twitter.com/Theme_Selection](https://twitter.com/Theme_Selection)
- Facebook : [https://www.facebook.com/ThemeSelections/](https://www.facebook.com/ThemeSelections/)
- Pinterest : [https://pinterest.com/themeselect/](https://pinterest.com/themeselect/)
- Instagram : [https://www.instagram.com/themeselection/](https://www.instagram.com/themeselection/)


CHANGELOG.md
<!-- Available h3 headings: Added, Fixed, Updated, Removed, Deprecated -->

# Changelog

All notable changes to this template will be documented in this file.

## v1.2.0 (2023-03-20)

### Added

- Added .gitattributes file
- Added Progress component page
- Added necessary props for the UserLayout
- Added custom variable `avatarBg` for Avatar background-color in the MUI's palette
- Added ability to access vertical or horizontal layout with menu without logging in
- Added articles on how to migrate from JWT Auth to [NextAuth](https://next-auth.js.org/), NextAuth with [CredentialsProvider](https://next-auth.js.org/configuration/providers/credentials) and NextAuth with [GoogleProvider](https://next-auth.js.org/providers/google) & [PrismaAdapter](https://next-auth.js.org/adapters/prisma)

### Updated

- Updated docs
- Updated FAQs on the Pricing page
- Changed MUI Links to Next.js Links
- Updated examples on the Dialog pages
- Updated `trackBg` colors in the theme
- Changed method of implementing semi-dark skin
- Updated input mask labels on the Input Mask page
- Updated bg-colors of social cards on the Card Basic page
- Updated MUI's theme structure which is in the `src/@core` folder
- Update `lang` attribute on the `<html>` tag when the language is changed
- Updated structure of points in the Knowledge Base section on the Help Center page
- Updated UI of avatar, breadcrumb, chip, dataGrid, list, tooltip, snackbar, switch components
- Updated all the packages (Refer to the official docs for the migration of `@mui/x-data-grid` from [here](https://mui.com/x/migration/migration-data-grid-v5/))

### Fixed

- Fixed types
- Fixed vertical menu re-renders on hover
- Fixed transition issue on pull-up Avatars
- Fixed Alert snackbar shadow in bordered skin
- Fixed shadows in the whole template when the skin is semi-dark
- Fixed `react-hot-toast`'s z-index when its position is on the left side
- Fixed shadow between nav header and nav items in the Vertical navigation menu in RTL
- Fixed bugs in the Chat, Calendar, Invoice, User apps and User Profile, FAQ, File Uploader pages

### Removed

- Removed unnecessary images
- Removed validation on the Register page
- Removed all the files and guides related to Firebase (in favour of NextAuth)

## v1.1.0 (2022-11-16)

### Added

- Added `.vscode` folder in all the `full-version` and `starter-kit` folders

### Updated

- Updated docs
- Updated Next.js to v13
- Optimized all the images
- Moved JWT configs to the `.env` file
- Updated spacing in the `Button` component
- Updated all the dependencies to the latest versions
- Updated all the MUI links (with `href='/'`) to Next.js links

### Fixed

- Fixed minor bugs
- Fixed border-right in the Calendar app
- Fixed console error on the Help Center page
- Fixed scroll issue when page loader renders
- Fixed spacing in the left sidebar in the Email app
- Fixed z-index of `react-hot-toast` in smaller screens
- Fixed a few re-renders in the Vertical navigation menu
- Fixed types in `UserLayout` component and custom components inside the `src/@core` folder

### Removed

- Removed validations from some pages

## v1.0.0 (2022-10-13)

### Added

- Initial Release


javascript-version/demo-configs/demo-2.js
/**
 * Config
 * -------------------------------------------------------------------------------------
 * ! IMPORTANT: Make sure you clear the browser local storage in order to see the config changes in the template.
 * ! To clear local storage, you may refer https://www.leadshook.com/help/how-to-clear-local-storage-in-google-chrome-browser/.
 */
const themeConfig = {
  // ** Layout Configs
  templateName: 'Sneat' /* App Name */,
  layout: 'vertical' /* vertical | horizontal */,
  mode: 'light' /* light | dark | semi-dark /*! Note: semi-dark value will only work for Vertical Layout */,
  direction: 'ltr' /* ltr | rtl */,
  skin: 'bordered' /* default | bordered */,
  contentWidth: 'boxed' /* full | boxed */,
  footer: 'static' /* fixed | static | hidden */,
  // ** Routing Configs
  routingLoader: true /* true | false */,
  // ** Navigation (Menu) Configs
  navHidden: false /* true | false */,
  menuTextTruncate: true /* true | false */,
  navSubItemIcon: 'bxs:circle' /* Icon */,
  verticalNavToggleType: 'accordion' /* accordion | collapse /*! Note: This is for Vertical navigation menu only */,
  navCollapsed: false /* true | false /*! Note: This is for Vertical navigation menu only */,
  navigationSize: 260 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  collapsedNavigationSize: 84 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  afterVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  beforeVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  horizontalMenuToggle: 'hover' /* click | hover /*! Note: This is for Horizontal navigation menu only */,
  horizontalMenuAnimation: true /* true | false */,
  // ** AppBar Configs
  appBar: 'fixed' /* fixed | static | hidden /*! Note: hidden value will only work for Vertical Layout */,
  appBarBlur: true /* true | false */,
  // ** Other Configs
  responsiveFontSizes: true /* true | false */,
  disableRipple: true /* true | false */,
  disableCustomizer: false /* true | false */,
  toastPosition: 'top-right' /* top-left | top-center | top-right | bottom-left | bottom-center | bottom-right */
}

export default themeConfig


javascript-version/demo-configs/demo-4.js
/**
 * Config
 * -------------------------------------------------------------------------------------
 * ! IMPORTANT: Make sure you clear the browser local storage in order to see the config changes in the template.
 * ! To clear local storage, you may refer https://www.leadshook.com/help/how-to-clear-local-storage-in-google-chrome-browser/.
 */
const themeConfig = {
  // ** Layout Configs
  templateName: 'Sneat' /* App Name */,
  layout: 'vertical' /* vertical | horizontal */,
  mode: 'dark' /* light | dark | semi-dark /*! Note: semi-dark value will only work for Vertical Layout */,
  direction: 'ltr' /* ltr | rtl */,
  skin: 'default' /* default | bordered */,
  contentWidth: 'boxed' /* full | boxed */,
  footer: 'static' /* fixed | static | hidden */,
  // ** Routing Configs
  routingLoader: true /* true | false */,
  // ** Navigation (Menu) Configs
  navHidden: false /* true | false */,
  menuTextTruncate: true /* true | false */,
  navSubItemIcon: 'bxs:circle' /* Icon */,
  verticalNavToggleType: 'accordion' /* accordion | collapse /*! Note: This is for Vertical navigation menu only */,
  navCollapsed: false /* true | false /*! Note: This is for Vertical navigation menu only */,
  navigationSize: 260 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  collapsedNavigationSize: 84 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  afterVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  beforeVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  horizontalMenuToggle: 'hover' /* click | hover /*! Note: This is for Horizontal navigation menu only */,
  horizontalMenuAnimation: true /* true | false */,
  // ** AppBar Configs
  appBar: 'fixed' /* fixed | static | hidden /*! Note: hidden value will only work for Vertical Layout */,
  appBarBlur: true /* true | false */,
  // ** Other Configs
  responsiveFontSizes: true /* true | false */,
  disableRipple: true /* true | false */,
  disableCustomizer: false /* true | false */,
  toastPosition: 'top-right' /* top-left | top-center | top-right | bottom-left | bottom-center | bottom-right */
}

export default themeConfig


javascript-version/demo-configs/demo-5.js
/**
 * Config
 * -------------------------------------------------------------------------------------
 * ! IMPORTANT: Make sure you clear the browser local storage in order to see the config changes in the template.
 * ! To clear local storage, you may refer https://www.leadshook.com/help/how-to-clear-local-storage-in-google-chrome-browser/.
 */
const themeConfig = {
  // ** Layout Configs
  templateName: 'Sneat' /* App Name */,
  layout: 'horizontal' /* vertical | horizontal */,
  mode: 'light' /* light | dark | semi-dark /*! Note: semi-dark value will only work for Vertical Layout */,
  direction: 'ltr' /* ltr | rtl */,
  skin: 'default' /* default | bordered */,
  contentWidth: 'boxed' /* full | boxed */,
  footer: 'static' /* fixed | static | hidden */,
  // ** Routing Configs
  routingLoader: true /* true | false */,
  // ** Navigation (Menu) Configs
  navHidden: false /* true | false */,
  menuTextTruncate: true /* true | false */,
  navSubItemIcon: 'bxs:circle' /* Icon */,
  verticalNavToggleType: 'accordion' /* accordion | collapse /*! Note: This is for Vertical navigation menu only */,
  navCollapsed: false /* true | false /*! Note: This is for Vertical navigation menu only */,
  navigationSize: 260 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  collapsedNavigationSize: 84 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  afterVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  beforeVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  horizontalMenuToggle: 'hover' /* click | hover /*! Note: This is for Horizontal navigation menu only */,
  horizontalMenuAnimation: true /* true | false */,
  // ** AppBar Configs
  appBar: 'fixed' /* fixed | static | hidden /*! Note: hidden value will only work for Vertical Layout */,
  appBarBlur: true /* true | false */,
  // ** Other Configs
  responsiveFontSizes: true /* true | false */,
  disableRipple: true /* true | false */,
  disableCustomizer: false /* true | false */,
  toastPosition: 'top-right' /* top-left | top-center | top-right | bottom-left | bottom-center | bottom-right */
}

export default themeConfig


javascript-version/demo-configs/demo-6.js
/**
 * Config
 * -------------------------------------------------------------------------------------
 * ! IMPORTANT: Make sure you clear the browser local storage in order to see the config changes in the template.
 * ! To clear local storage, you may refer https://www.leadshook.com/help/how-to-clear-local-storage-in-google-chrome-browser/.
 */
const themeConfig = {
  // ** Layout Configs
  templateName: 'Sneat' /* App Name */,
  layout: 'horizontal' /* vertical | horizontal */,
  mode: 'dark' /* light | dark | semi-dark /*! Note: semi-dark value will only work for Vertical Layout */,
  direction: 'ltr' /* ltr | rtl */,
  skin: 'bordered' /* default | bordered */,
  contentWidth: 'boxed' /* full | boxed */,
  footer: 'static' /* fixed | static | hidden */,
  // ** Routing Configs
  routingLoader: true /* true | false */,
  // ** Navigation (Menu) Configs
  navHidden: false /* true | false */,
  menuTextTruncate: true /* true | false */,
  navSubItemIcon: 'bxs:circle' /* Icon */,
  verticalNavToggleType: 'accordion' /* accordion | collapse /*! Note: This is for Vertical navigation menu only */,
  navCollapsed: false /* true | false /*! Note: This is for Vertical navigation menu only */,
  navigationSize: 260 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  collapsedNavigationSize: 84 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  afterVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  beforeVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  horizontalMenuToggle: 'hover' /* click | hover /*! Note: This is for Horizontal navigation menu only */,
  horizontalMenuAnimation: true /* true | false */,
  // ** AppBar Configs
  appBar: 'fixed' /* fixed | static | hidden /*! Note: hidden value will only work for Vertical Layout */,
  appBarBlur: true /* true | false */,
  // ** Other Configs
  responsiveFontSizes: true /* true | false */,
  disableRipple: true /* true | false */,
  disableCustomizer: false /* true | false */,
  toastPosition: 'top-right' /* top-left | top-center | top-right | bottom-left | bottom-center | bottom-right */
}

export default themeConfig


javascript-version/demo-configs/demo-3.js
/**
 * Config
 * -------------------------------------------------------------------------------------
 * ! IMPORTANT: Make sure you clear the browser local storage in order to see the config changes in the template.
 * ! To clear local storage, you may refer https://www.leadshook.com/help/how-to-clear-local-storage-in-google-chrome-browser/.
 */
const themeConfig = {
  // ** Layout Configs
  templateName: 'Sneat' /* App Name */,
  layout: 'vertical' /* vertical | horizontal */,
  mode: 'semi-dark' /* light | dark | semi-dark /*! Note: semi-dark value will only work for Vertical Layout */,
  direction: 'ltr' /* ltr | rtl */,
  skin: 'default' /* default | bordered */,
  contentWidth: 'boxed' /* full | boxed */,
  footer: 'static' /* fixed | static | hidden */,
  // ** Routing Configs
  routingLoader: true /* true | false */,
  // ** Navigation (Menu) Configs
  navHidden: false /* true | false */,
  menuTextTruncate: true /* true | false */,
  navSubItemIcon: 'bxs:circle' /* Icon */,
  verticalNavToggleType: 'accordion' /* accordion | collapse /*! Note: This is for Vertical navigation menu only */,
  navCollapsed: false /* true | false /*! Note: This is for Vertical navigation menu only */,
  navigationSize: 260 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  collapsedNavigationSize: 84 /* Number in px(Pixels) /*! Note: This is for Vertical navigation menu only */,
  afterVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  beforeVerticalNavMenuContentPosition: 'fixed' /* fixed | static */,
  horizontalMenuToggle: 'hover' /* click | hover /*! Note: This is for Horizontal navigation menu only */,
  horizontalMenuAnimation: true /* true | false */,
  // ** AppBar Configs
  appBar: 'fixed' /* fixed | static | hidden /*! Note: hidden value will only work for Vertical Layout */,
  appBarBlur: true /* true | false */,
  // ** Other Configs
  responsiveFontSizes: true /* true | false */,
  disableRipple: true /* true | false */,
  disableCustomizer: false /* true | false */,
  toastPosition: 'top-right' /* top-left | top-center | top-right | bottom-left | bottom-center | bottom-right */
}

export default themeConfig
